simulator/mips/mips_instr.cpp

```diff
@@ -294,6 +294,44 @@ BaseMIPSInstr<RegisterUInt>::BaseMIPSInstr( MIPSVersion version, uint32 bytes, A
     }
 }
 
+template<typename RegisterUInt>
+BaseMIPSInstr<RegisterUInt>::BaseMIPSInstr( MIPSVersion version, std::string_view str_opcode, Addr PC) :
+    new_PC( PC + 4),
+    PC( PC)
+{
+    auto it = find_entry( isaMapR,      str_opcode);
+    if ( it == isaMapR.end())
+        it =  find_entry( isaMapRI,     str_opcode);
+    if ( it == isaMapRI.end())
+        it =  find_entry( isaMapMIPS32, str_opcode);
+    if ( it == isaMapMIPS32.end())
+        it =  find_entry( isaMapIJ,     str_opcode);
+    if ( it == isaMapIJ.end())
+    {
+        std::ostringstream oss;
```

Comment:
The block is duplicated from decoding ctor, please add a separate method

Reply:
This block is similar, but uses stringed opcode. Should I add another method for the class (opcode and str_opcode variants)?

Reply:
Ok, I see the difference now. Let's leave it as is.